Duplicati/Library/Backend/pCloud/pCloudBackend.cs

```diff
@@ -0,0 +1,544 @@
+    /// requests
+    /// </summary>
+    private ulong? _CachedFolderID;
+
+    private const string AUTHENTICATION_OPTION = "authid";
+
+    private static readonly Dictionary<string, string> PCLOUD_SERVERS = new(StringComparer.OrdinalIgnoreCase)
+    {
+        { "pCloud Global", "api.pcloud.com" },
+        { "pCloud (EU)", "eapi.pcloud.com" },
+    };
+
+    /// <summary>
+    /// Empty constructor is required for the backend to be loaded by the backend factory
+    /// </summary>
+    public pCloudBackend()
+    {
+    }
+
+    /// <summary>
+    /// Actual constructor for the backend that accepts the url and options
+    /// </summary>
+    /// <param name="url">URL in Duplicati Uri format</param>
+    /// <param name="options">options to be used in the backend</param>
+    public pCloudBackend(string url, Dictionary<string, string> options)
+    {
+        var uri = new Utility.Uri(url);
+        uri.RequireHost();
+        _DnsName = uri.Host;
+
+        if (options.TryGetValue(AUTHENTICATION_OPTION, out var option))
+            _Token = option;
+
+        if (!PCLOUD_SERVERS.ContainsValue(uri.Host))
+            throw new UserInformationException(Strings.pCloudBackend.InvalidServerSpecified,
+                "InvalidpCloudServerSpecified");
+
+        if (string.IsNullOrWhiteSpace(uri.Host))
+            throw new UserInformationException(Strings.pCloudBackend.NoServerSpecified, "NopCloudServerSpecified");
+
+        _Path = uri.Path;
+        _ServerUrl = uri.Host;
+    }
+
+    /// <summary>
+    /// Implementation of interface property to return supported command parameters
+    /// </summary>
+    public IList<ICommandLineArgument> SupportedCommands
+    {
+        get
+        {
+            return new List<ICommandLineArgument>(new ICommandLineArgument[]
+            {
+                new CommandLineArgument(AUTHENTICATION_OPTION,
+                    CommandLineArgument.ArgumentType.Password,
+                    Strings.pCloudBackend.AuthPasswordDescriptionShort,
+                    Strings.pCloudBackend.AuthPasswordDescriptionLong),
+            });
+        }
+    }
+
+    /// <summary>
+    /// Implementation of interface method for listing remote folder contents.
+    /// The root parameter is used to list the root folder, as the pCloud API
+    /// when using oauth tokens creates an isolated folder Applications/ApplicationName
+    ///
+    /// When root = true, this is the directory it will be listed.
+    /// When root = false, it will list the _Path directory if it exists, which is
+    /// where the Duplicati backup files are stored.
+    /// </summary>
+    /// <param name="root">True to list absolute root, false to list _Path directory</param>
+    /// <returns></returns>
+    private IEnumerable<IFileEntry> List(bool root)
+    {
+        if (_CachedFolderID == null && string.IsNullOrWhiteSpace(_Path) == false && root == false)
+            CreateFolderAsync(new CancellationToken()).GetAwaiter().GetResult();
```

Comment:
For async-to-sync mapping, this should use `.Await()` (extension method) so all are working the same way.
Instead of `new CancellationToken()` this should be `CancellationToken.None`.